### PR TITLE
[WIP] Refactoring of Docker-PHP

### PR DESCRIPTION
**Pull request by joelwurtz**
Hey,

This is a pretty long PR, and a work of several months on many libraries.

To know how it works see my last post on #46 

TL; DR; We know just have to maintain the swagger specification and a tiny layer around the generated code to have a better user experience, but there in this PR a 100% coverage of all docker calls.

You can test this at your own risk.

For the moment this is completly BC with the old library, i don't think we will provide some sort of compatibility layer unless someone want to write it.

This should also fix other issue like #139, #122, #96, #30, #138 and #108 

There is still some works to do on the jane/swagger library to handle incoming object to be serialized in the request.

I'm also wondering about moving this under the `Engine` namespace, so we can add registry, index and others api from Docker WDYT @ubermuda ?

### CODE REVIEW

**Comment by joelwurtz** [comment]
So tests are green :)

Incoming object are now correctly serialized with last update from the jane swagger library.

Missing parts : 

 - [ ] Stable version of php-http/httplug and others components
 - [ ] Stable version of jane-swagger
 - [ ] Friendly experience layer (interactive run, using context builder, interactive build, ....)
 - [ ] More tests

**Comment by joelwurtz** [comment]
poke @hacfi, don't know if you are still using docker-php, will be glad to have some inputs for this.

**Comment by hacfi** [comment]
@joelwurtz Will review it on the weekend. Generally I really like the concept of Jane!
